#! /bin/sh /usr/share/dpatch/dpatch-run
## svn-r4555_fix-crash-on-icon-size-change.dpatch by Chris Coulson <chrisccoulson@googlemail.com>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Patch backported from SVN trunk revisions 4555 and 4556
## DP: Launchpad bug: https://bugs.launchpad.net/f-spot/+bug/291189
## DP: Gnome bug: http://bugzilla.gnome.org/show_bug.cgi?id=557793

@DPATCH@
diff -urNad f-spot-0.5.0.3~/src/Core/Tag.cs f-spot-0.5.0.3/src/Core/Tag.cs
--- f-spot-0.5.0.3~/src/Core/Tag.cs	2008-08-31 09:52:18.000000000 +0100
+++ f-spot-0.5.0.3/src/Core/Tag.cs	2008-11-04 17:39:37.000000000 +0000
@@ -108,6 +108,9 @@
 					if (Math.Max (cached_icon.Width, cached_icon.Height) <= (int) tag_icon_size) 
 						return cached_icon;
 				}
+				if (icon == null)
+					return null;
+					
 				if (Math.Max (icon.Width, icon.Height) >= (int) tag_icon_size) { //Don't upscale
 					if (cached_icon != null)
 						cached_icon.Dispose ();
diff -urNad f-spot-0.5.0.3~/src/TagSelectionWidget.cs f-spot-0.5.0.3/src/TagSelectionWidget.cs
--- f-spot-0.5.0.3~/src/TagSelectionWidget.cs	2008-09-26 09:27:39.000000000 +0100
+++ f-spot-0.5.0.3/src/TagSelectionWidget.cs	2008-11-04 17:37:19.000000000 +0000
@@ -192,7 +192,7 @@
 		SetBackground (renderer, tag);
 
 		// FIXME I can't set the Pixbuf to null, not sure if it's a GTK# bug...
-		if (tag.Icon != null) {
+		if (tag.SizedIcon != null) {
 			if (FSpot.ColorManagement.IsEnabled) {
 				//FIXME
 				Gdk.Pixbuf temp = tag.SizedIcon.Copy();
